src/decaylanguage/dec/dec.py

```diff
@@ -687,38 +708,69 @@ def print_decay_modes(
         ascending: bool, optional, default=False
             Print the list of decay modes ordered in ascending/descending order
             of branching fraction.
+        normalize: bool, optional, default=True
+            Print the branching fractions normalized to unity
+            (this does not affect the values parsed and actually stored in memory).
         """
         if pdg_name:
             mother = PDG2EvtGenNameMap[mother]
 
         dms = self._find_decay_modes(mother)
 
-        ls = []
+        ls_dict = dict()
         for dm in dms:
+            bf, fsp_names, model, model_params = self._decay_mode_details(dm, **kwargs)
+            model_params = [str(i) for i in model_params]
+            ls_dict[bf] = (fsp_names, model, model_params)
+
+        dec_details = list(ls_dict.values())
+        ls_attrs_aligned = list(
+            zip_longest(
+                *[self._align_items(i) for i in zip(*dec_details)], fillvalue=""
+            )
+        )
+
+        ls = [(bf, ls_attrs_aligned[idx]) for idx, bf in enumerate(ls_dict)]
+        ls.sort(key=operator.itemgetter(0), reverse=(not ascending))
+        norm = sum(bf for bf, _ in ls) if normalize else 1
+
+        for bf, info in ls:
             if print_model:
-                dm_details = self._decay_mode_details(dm)
-                ls.append(
-                    (
-                        dm_details[0],
-                        (
-                            "%-50s %15s %s"
-                            % (
-                                "  ".join(dm_details[1]),
-                                dm_details[2],
-                                dm_details[3],
-                            )
-                        ),
-                    )
-                )
+                line = "  {:.4f}   {}     {}  {}".format(bf / norm, *info)
+            else:
+                line = "  {:.4f}   {}".format(bf / norm, info[0])
+            print(line.rstrip() + ";")
 
+    @staticmethod
```

Comment:
What is the reason to make this a static method? The internal function is so "internal" that one might even consider having in inside `print_decay_modes`.

Reply:
Two reasons:

1. Reduce the number of lines for `print_decay_modes` (I think shorter functions are easier to read)
2. This alignment could be useful in somewhere else, and making it a separate static method makes unit-test it easier, IMO.

Reply:
OK, let's go with that, then. No harm anyway.